### DISCUSSION TITLE
HDR 显示模式下 SDR 内容灰暗的原因

### DISCUSSION THREAD

**Discussion by cy0o0** [dyphire/mpv-config#156] [General]
如题，请教下大佬该如何解决这个问题

**Comment by dyphire** [comment]
~~心理作用，无论是本仓库配置还是 mpv 上游都没有进行相关更改~~

**编辑：** 具体原因及修复方法见 https://github.com/dyphire/mpv-config/discussions/156#discussioncomment-13190743

**Comment by cy0o0** [comment]
![image](https://github.com/user-attachments/assets/94e01999-809e-431d-8ffe-f790651a145f)
这个是我在我的电脑上下载两个版本，播放同一视频文件时的截图，左为12.04，右为10.07，应该不是我的心理作用吧。

**Reply by ride-donkey** [reply]
你的显示器有使用hdr吗 我的新版mpv在hdr显示器上的全局亮度明显低很多（包括ui，sdr显示器一切正常）然后通过调整hdr的一堆设置把srd视频拉到能看的地步（但是本质上是不太正确的亮度映射）
然后字幕依旧是非常的暗淡，看的很难受

**Reply by nanachisol** [reply]
> 你的显示器有使用hdr吗 我的新版mpv在hdr显示器上的全局亮度明显低很多（包括ui，sdr显示器一切正常）然后通过调整hdr的一堆设置把srd视频拉到能看的地步（但是本质上是不太正确的亮度映射） 然后字幕依旧是非常的暗淡，看的很难受

https://github.com/natural-harmonia-gropius/hdr-toys?tab=readme-ov-file
https://github.com/mpv-player/mpv/discussions/14757
试试这两个吧，没有HDR显示器，无法帮你测试

**Reply by wathper** [reply]
新版本我在hdr显示器上没有观察到亮度有明显变化。

**Comment by cy0o0** [comment]
我在下载后没有修改任何配置，所以不太清楚如何排查该问题产生原因，大佬能给些建议吗

**Comment by nanachisol** [comment]
> ![image](https://private-user-images.githubusercontent.com/52368412/397964890-94e01999-809e-431d-8ffe-f790651a145f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4NDU2MTAsIm5iZiI6MTczNDg0NTMxMCwicGF0aCI6Ii81MjM2ODQxMi8zOTc5NjQ4OTAtOTRlMDE5OTktODA5ZS00MzFkLThmZmUtZjc5MDY1MWExNDVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIyVDA1MjgzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxOWUwNTk2MWE5M2VjNDIwNmI0NzA2YjcyMTRhM2RmNDlkZDBiNWI0YWQ2NGI2MmNjNWIyYjI2MDJlNWJhYzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HDsSSVkPcOVhFVlpHwqdmdC67N3lXsv9FEtSRhLR7Os) 这个是我在我的电脑上下载两个版本，播放同一视频文件时的截图，左为12.04，右为10.07，应该不是我的心理作用吧。

是不是与影片格式有关。
先用https://github.com/zhongfly/mpv-winbuild/releases/tag/2024-12-21-d827019 最新git版测试一下
其次如果--no-config也能复现，可以到上游提issue
ps:官网倒是有相似的问题，不过一般都是hdr相关的，样本这种老番也有HDR吗？
https://github.com/mpv-player/mpv/issues/15138
https://github.com/mpv-player/mpv/issues/15107
https://github.com/mpv-player/mpv/issues/14943

**Comment by dyphire** [comment]
> ![image](https://private-user-images.githubusercontent.com/52368412/397964890-94e01999-809e-431d-8ffe-f790651a145f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4NjQ3MDMsIm5iZiI6MTczNDg2NDQwMywicGF0aCI6Ii81MjM2ODQxMi8zOTc5NjQ4OTAtOTRlMDE5OTktODA5ZS00MzFkLThmZmUtZjc5MDY1MWExNDVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIyVDEwNDY0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc3ZDgwZjM5NTZiMzNjYWQ5NWY5YjJmZWRjNTlmYWZkYWUxZjViMTExODBlYzk5NTA5MGIxMGRiODc3NTcwMzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UbyV7S_pYvGqX9rssO0zFDHB4gLhWaHcOnFlL2EdSCM)
> 这个是我在我的电脑上下载两个版本，播放同一视频文件时的截图，左为12.04，右为10.07，应该不是我的心理作用吧。

无法复现，只能推测可能跟显卡驱动和系统设置有关

**Comment by nanachisol** [comment]
> ![image](https://private-user-images.githubusercontent.com/52368412/397964890-94e01999-809e-431d-8ffe-f790651a145f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4Njg2NjAsIm5iZiI6MTczNDg2ODM2MCwicGF0aCI6Ii81MjM2ODQxMi8zOTc5NjQ4OTAtOTRlMDE5OTktODA5ZS00MzFkLThmZmUtZjc5MDY1MWExNDVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIyVDExNTI0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2NjFiZjQ2YWQwZTA4NWVjMjlhZTBiOWIyNTFjMzYwMWU0NzczMjNmYzJkYTUwY2Q1MzJlZWU4MTE0OWJmZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fOihRY4tBqc2AW50GvEvld4LdbEZMniKiYhZ3OX7zAw) 这个是我在我的电脑上下载两个版本，播放同一视频文件时的截图，左为12.04，右为10.07，应该不是我的心理作用吧。


无法复现，以下压缩包为1007与1204截图，虽然不是同一帧，但亮度无法观测出差别。能否提供sample files
[【もう少しの夢】乃木坂46 4th YEAR BIRTHDAY LIVE 2016. 4K ver. 西野七瀨 [BkAmR1Pt9hc].zip](https://github.com/user-attachments/files/18221749/46.4th.YEAR.BIRTHDAY.LIVE.2016.4K.ver.BkAmR1Pt9hc.zip)

**Comment by cy0o0** [comment]
通过替换不同的mpv.exe，测试了一下，应该是mpv的问题，之后版本的mpv不知道为什么在我的环境中就有这个问题。

**Comment by dyphire** [comment]
> 通过替换不同的mpv.exe，测试了一下，应该是mpv的问题，之后版本的mpv不知道为什么在我的环境中就有这个问题。

在你环境中测试的最后一个正常工作的 mpv 版本是哪个？

**Comment by cy0o0** [comment]
https://github.com/dyphire/mpv-winbuild/releases/tag/mpv_own-2024-11-24 我使用该版本及以后会出现该情况

**Comment by dyphire** [comment]
> https://github.com/dyphire/mpv-winbuild/releases/tag/mpv_own-2024-11-24 我使用该版本及以后会出现该情况

推测与此 pr 有关：https://github.com/mpv-player/mpv/pull/15279

**Comment by verygoodlee** [comment]
可以尝试下添加这个选项`target-colorspace-hint=no`，在旧版这本来是默认值的，后来改成yes，然后又改成了auto，

在默认值yes的那段时间我碰到过类似的情况，不过只在HDR视频中有影响，我Windows设置中甚至都没开显示器HDR，选项也生效了，
默认值变为auto后我也一直用的no

**Comment by nanachisol** [comment]
https://github.com/mpv-player/mpv/pull/15820

**Comment by dyphire** [comment]
在最新发布版中应该不再出现

**Comment by dyphire** [comment]
最近购入了一块 HDR 显示器，然后我在系统的 HDR 显示模式下测试并复现了此问题

让我们来品鉴一下 mpv 的抽象行为，以下为显示器 HDR 显示模式下 mpv 播放 SDR 内容的测试表现：

|  mpv --vo=gpu-next（显示器 sdr 模式）  |   mpv --vo=gpu-next --target-colorspace-hint=no（显示器 hdr 模式）  |  mpv --vo=gpu-next --target-colorspace-hint=no --target-peak=203（显示器 hdr 模式）  |  mpc-be（显示器 hdr 模式）  |
| ---------------- | ---------------- | ---------------- |  ---------------- |
| ![2025-05-19 161000](https://github.com/user-attachments/assets/e72bad2a-5f3c-4c55-952d-f1b437ae15ca) | ![2025-05-19 161023](https://github.com/user-attachments/assets/c2e9f19b-1222-44a4-8c2a-c2388e796c65) |  ![2025-05-19 161057](https://github.com/user-attachments/assets/6e812244-12a7-4cdb-a0aa-0f4e6ef3baea) |  ![2025-05-19 161324](https://github.com/user-attachments/assets/b51060dc-8b73-4e4e-8d40-07a033f9ba2d)  |

以下是 OSD 中的统计信息
|  mpv --vo=gpu-next（显示器 sdr 模式）  |   mpv --vo=gpu-next --target-colorspace-hint=no（显示器 hdr 模式）  |  mpv --vo=gpu-next --target-colorspace-hint=no --target-peak=203（显示器 hdr 模式）  |
| ---------------- | ---------------- | ---------------- |
| ![2025-05-19 161138](https://github.com/user-attachments/assets/34fc09e2-4013-4c41-8c78-d97629bcba1d) |  ![2025-05-19 161210](https://github.com/user-attachments/assets/26aa19a1-8e27-4c94-ac4f-8f5dbe5ac999) |  ![2025-05-19 161232](https://github.com/user-attachments/assets/d9335cb6-7d8f-49c5-aaa9-93edc28431a6)  |

针对 mpv 的 --target-peak 选项默参的抽象行为，在最新提交中予以修复 https://github.com/dyphire/mpv-config/commit/37e46d0

**Reply by dyphire** [reply]
已在 mpv 上游进行反馈：https://github.com/mpv-player/mpv/issues/16360#issuecomment-2891038681

**Reply by dyphire** [reply]
mpv 开发人员似乎认为这是 feature：https://github.com/mpv-player/mpv/issues/15411#issuecomment-2508957305
属实难绷，一众播放器里只有 mpv 有这种抽象行为，其他播放器表现正常

**Comment by dyphire** [comment]
以下是补充的对比测试：
|  mpv --vo=gpu-next --target-colorspace-hint=auto（显示器 hdr 模式）  |   mpv --vo=gpu-next --gpu-api=vulkan --target-colorspace-hint=no（显示器 hdr 模式）  | 
| ---------------- | ---------------- |
| ![2025-05-19 193758](https://github.com/user-attachments/assets/e8eddde8-5256-4681-9543-6c8746137e7a) | ![2025-05-19 193840](https://github.com/user-attachments/assets/a3d9141e-26c9-44bd-a156-e7eca398bf33) | 

OSD 中的统计信息

|  mpv --vo=gpu-next --target-colorspace-hint=auto（显示器 hdr 模式）  |   mpv --vo=gpu-next --gpu-api=vulkan --target-colorspace-hint=no（显示器 hdr 模式）  | 
| ---------------- | ---------------- |
| ![2025-05-19 193805](https://github.com/user-attachments/assets/e0ffa7fc-22fe-4cf0-8ec4-62c0ff21ab78)  | ![2025-05-19 193851](https://github.com/user-attachments/assets/3c6b64eb-eb9c-4855-a91e-deace8d548ea)  | 

可以明确的是引入此 bug 的就是上面提到的 https://github.com/mpv-player/mpv/pull/15279

**Comment by dyphire** [comment]
鉴于上游迟迟未对此做出回应，我提了一个 pr 进行修复：https://github.com/mpv-player/mpv/pull/16417

**Reply by dyphire** [reply]
看来是没希望在上游得到修复了，开发者还是固执地认为这一行为符合预期

**Reply by dyphire** [reply]
由 https://github.com/mpv-player/mpv/pull/16502 取代

**Comment by dyphire** [comment]
上游 PR https://github.com/mpv-player/mpv/pull/16502 已合并，关闭此讨论
